### ISSUE TITLE
use mybinder to make the tutorial runnable

### ISSUE THREAD

**Issue by gdementen** [liam2/larray#73]
http://mybinder.org/

**Comment by alixdamman** [comment]
I tried. It doesn't work on a specific branch, you can build an image of the branch master only.

**Comment by gdementen** [comment]
Yes, it would be nicer to be able to build from a (version) tag instead of master but that does not prevent us to use it, right?

FWIW, the corresponding issue is at: https://github.com/binder-project/binder/issues/31

**Comment by gdementen** [comment]
The project has greatly improved in the past few months (they just release version2). It can now build from a specific tag or branch and only for a specific notebook and it is incredibly fast. We only need to have a notebook and an environment file and that's it. Ideally I would like the tutorial to be runnable. The only question is thus: how to keep a notebook in sync with the tutorial.rst file in the documentation AND keep the ability to review/diff the changes to the tutorial?

**Comment by alixdamman** [comment]
@gdementen 

> The only question is thus: how to keep a notebook in sync with the tutorial.rst file in the documentation AND keep the ability to review/diff the changes to the tutorial?

We could convert (once again) our tutorial into jupyter notebook(s) and use the command line tool `jupyter nbconvert tutorial.ipynb --to rst` to have a RST file version of the tutorial which would be more easier to review. The main drawback is obviously to forget to convert the notebook(s) into RST file(s) during a PR.

What do you think?

**Comment by gdementen** [comment]
or we use https://github.com/prabhuramachandran/ipyaml

**Comment by alixdamman** [comment]
@gdementen  You are suggesting to use https://github.com/prabhuramachandran/ipyaml because it automatically load and save the notebook as YAML file or because you want to use the YAML format specifically?

Another possibility is to use a post-save hook to save as RST file : http://jupyter-notebook.readthedocs.io/en/stable/extending/savehooks.html

**Comment by gdementen** [comment]
I don't care about yaml specifically but I do care about the ability to review and merge changes/pull requests made to the tutorial. This should be much better with the .yml files than with .ipynb files.

The .rst file is another option which solves the review part but the yaml solution should have the additional advantage that you can **merge** conflicting changes, which is impossible AFAIK with .ipynb files (and would be hard/impractical to do with the .rst because the .rst is not the source file). 

But this all assumes ipyaml actually works nicely which I don't know since I have not tried it yet.
